vunit.dockerfile

```diff
@@ -1,28 +1,49 @@
 ARG TAG="buster-mcode"
 
+#---
+
 FROM ghdl/ghdl:$TAG AS base
+
 ARG PY_PACKAGES
-RUN apt-get update -qq \
+
+RUN set -ux \
+ && apt-get update -qq \
  && DEBIAN_FRONTEND=noninteractive apt-get -y install --no-install-recommends \
-    ca-certificates \
-    curl \
-    make \
     python3 \
     python3-pip \
  && apt-get autoclean && apt-get clean && apt-get -y autoremove \
- && update-ca-certificates \
- && rm -rf /var/lib/apt/lists/* \
- && pip3 install -U pip setuptools wheel $PY_PACKAGES
+ && rm -rf /var/lib/apt/lists/*
+
+RUN set -ux \
+ && pip3 install --upgrade setuptools wheel $PY_PACKAGES \
+ && rm -rf ~/.cache
+
+#---
 
 FROM base as stable
-RUN pip3 install vunit_hdl
 
-FROM alpine as get-master
-RUN apk add --no-cache --update git && git clone --recurse-submodules https://github.com/VUnit/vunit /tmp/vunit
+RUN set -ux \
+ && pip3 install vunit_hdl \
+ && rm -rf ~/.cache \
+ # Make sure installation worked
+ && python3 -c "import vunit; print(vunit.__version__)"
```

Comment:
Maybe we want to make this a test in CI (after building the image but before pushing it)?

Reply:
I've only just started doing things like this in docker images. But I think it is extremely neat, since you get instant feedback whether it works or not, directly when building the image. Regardless of how you build an image (with a CI script or manually). 

In this case, if something went wrong with the VUnit pip installation, the build of the image would fail (since the python call would return non  zero), and a push would not occur.

Having a sanity-check like this in the Dockerfile, we would have found e.g. #31 straight away since the build would have failed. I think that having a small sanity-check at the end of a build step is a very good idea, that can be applied in a lot of places.

Reply:
I agree with adding a sanity-check that prevents images from being uploaded. My concern is about including it in the dockerfile. This test is rather naive, but adding any "proper" test would involve retrieving some example/testsuite and running it. That would be cumbersome for ensuring that no trash is left behind.

See, for example, any of the jobs in workflow 'test': https://github.com/ghdl/docker/runs/1393531346?check_suite_focus=true#step:4:986.

- GHDL is built in a container.
- The package is installed in an image through a Dockerfile.
- A container is used for running the tests.
- If successful, the image is pushed.

So, my proposal would be to add the following command after L187: https://github.com/ghdl/docker/blob/master/run.sh#L188

```sh
docker run --rm ghdl/$DREPO:$DTAG python3 -c "import vunit; print(vunit.__version__)"
```

In the mid term, I want to have a test script for each tool, because the single script approach in this repo does not scale well.

Reply:
I agree that the method of doing test in the dockerfile only makes sense for very small sanity-checks. For proper tests, the burden of having to clean up and not leave a footprint is too great. 

I have removed the sanity check from Dockerfile, and added the `docker run` call to run.sh per your suggestion.